### ISSUE TITLE
MySQL Bit flag not a good fit with BINARY

### ISSUE THREAD

**Issue by gpit2286** [diesel-rs/diesel#1995]
## Setup
* MySQL database. 
* Table that contains a BIT field. 

### Versions

- **Rust:** 1.33.0-nightly
- **Diesel:** 1.4.1
- **Database:** MySQL
- **Operating System** MacOS 

### Feature Flags

- **diesel:** [chrono, mysql] 

## Problem Description
Currently the BIT field is associated with the binary type.  For using the FromSql, there is not much that is a bother except for it puts the bit in a [u8].  However, filtering a bit field seems to be more complicated.  In order to use the ToSql on a binary field, you need to use a Vec<u8> or a u8 slice.  When diesel parses this, it turns it into an array which will never match a single bit.   

### What are you trying to accomplish?
It would be great if we moved the MySQL bit flag away from the binary data type and associate it with a bool value in order to filter the bit field in queries properly.  I would love to avoid using the raw SQL patch I have in there now (`sql("ActiveFlag = 1"`)

### Are you seeing any additional errors?
No 

### Steps to reproduce
* Build MySQL table with bit field. 
* Create schema for table 
* Try to use: 

```
table.filter(bit_field.eq(vec![1].to_sql::<Binary>())
```

Actual query will be similar to (Notice array in binds parameter): 
```
Query { sql: "SELECT `paymentterms`.`id`, `paymentterms`.`accountingHash`, `paymentterms`.`accountingId`, `paymentterms`.`activeFlag`, `paymentterms`.`dateCreated`, `paymentterms`.`dateLastModified`, `paymentterms`.`defaultTerm`, `paymentterms`.`discount`, `paymentterms`.`discountDays`, `paymentterms`.`name`, `paymentterms`.`netDays`, `paymentterms`.`nextMonth`, `paymentterms`.`readOnly`, `paymentterms`.`typeId` FROM `paymentterms` WHERE `paymentterms`.`activeFlag` = ?", binds: [[1]] }
```

## Checklist

- [x ] I have already looked over the [issue tracker](https://github.com/diesel-rs/diesel/issues) for similar issues.
- [ x] This issue can be reproduced on Rust's stable channel. (Your issue will be
  closed if this is not the case)

**Comment by sgrif** [comment]
We could consider this for 2.0. `BIT` and `BIT(1)` could be inferred as `Boolean` instead of `Binary`. We'd need to continue to infer `BIT(n)` to be `Binary` for n > 1

**Comment by gpit2286** [comment]
For anyone who is looking to do the same thing, Below is my code. 

```rust 
use diesel::deserialize::{self, FromSql};
use diesel::mysql::Mysql;
use diesel::serialize::{self, IsNull, Output, ToSql};
use diesel::sql_types::Binary;
use diesel::*;
use std::io::Write;
use serde::{Serialize};

#[derive(Debug, PartialEq, FromSqlRow, AsExpression, Serialize)]
#[sql_type = "Binary"]
pub struct BitBool(bool);

impl ToSql<Binary, Mysql> for BitBool {
    fn to_sql<W: Write>(&self, out: &mut Output<W, Mysql>) -> serialize::Result {
        match *self {
            BitBool(true) => out.write_all(&[1])?,
            BitBool(_) => out.write_all(&[0])?
        };
        Ok(IsNull::No)
    }
}

impl FromSql<Binary, Mysql> for BitBool {
    fn from_sql(bytes: Option<&[u8]>) -> deserialize::Result<Self> {
        let r = not_none!(bytes).iter().any(|x| *x != 0);
        Ok(BitBool(r))
    }
}

table! {
    test (id) {
        id -> Integer,
        active -> Bit
    }
}

struct TextRow {
    id: i32, 
    active: BitBool
}
```

**Comment by weiznich** [comment]
Fixed by #2290
